### PR TITLE
[FIX] Misaligned username on Room Info card for omnichannel chats

### PR DESCRIPTION
**Pull request by murtaza98**
## Issue(s)

image

### CODE REVIEW

**Review by dougfabris — CHANGES_REQUESTED** [review]
@murtaza98 I strongly recommend you replace UserCard.Username where Omnichannel is using it incorrectly:
- `apps/meteor/client/views/omnichannel/directory/calls/contextualBar/VoipInfo.tsx`
- `apps/meteor/client/views/omnichannel/directory/chats/contextualBar/AgentField.js`
- `apps/meteor/client/views/omnichannel/directory/chats/contextualBar/ContactField.js`
- `apps/meteor/client/views/omnichannel/directory/contacts/contextualBar/ContactInfo.js`
- `apps/meteor/ee/client/omnichannel/ContactManagerInfo.js`

**Comment by murtaza98** [comment]
Hey, @dougfabris Thanks for the review!! I've made the changes you requested. Do you mind taking a look at it again?

**Review by dougfabris — CHANGES_REQUESTED** [review]
@murtaza98 `UserCard.Username` should be used only inside `UserCard` 😢

**Review by ggazzo — CHANGES_REQUESTED** [review]
`UserCard.Username` is a component developed and meant to be used inside `UserCard`

this usage is wrong and there is no guarantee that tomorrow we wont change/remove or do whatever we want without consider outside usages

**Comment by ggazzo** [comment]
[FIX] ;)

**Comment by murtaza98** [comment]
Thanks for the review guys. I've now removed the usage of `UserCard.Username` component from all the omnichannel components & created an instead created a similar new component for omnichannel usage. Please review it again when you get a chance. Thanks!

File: apps/meteor/client/components/Omnichannel/AgentOrContactCardUsername.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Box } from '@rocket.chat/fuselage';
```
Comment (dougfabris) [inline apps/meteor/client/components/Omnichannel/AgentOrContactCardUsername.tsx:1]: I suggest you move this component to `apps/meteor/client/views/omnichannel/components`

**Review by dougfabris — CHANGES_REQUESTED** [review]
I still missing the replacement on this file: `apps/meteor/client/views/omnichannel/directory/calls/contextualBar/VoipInfo.tsx`

**Comment by murtaza98** [comment]
> I still missing the replacement on this file: `apps/meteor/client/views/omnichannel/directory/calls/contextualBar/VoipInfo.tsx`

Oops! Missed that one. It should now be fixed

**Review by ggazzo — DISMISSED** [review]
There is no card at this component, the component name is weird

**Comment by murtaza98** [comment]
> There is no card at this component, the component name is weird

Yeah, I agree since this isn't a card. I have removed it now. Thanks :)

**Comment by KevLehman** [comment]
So this was a fix after all 😂

File: apps/meteor/ee/client/omnichannel/ContactManagerInfo.js [hunk 2]
```diff
@@ -24,13 +25,15 @@ function ContactManagerInfo({ username }) {
...
-					({name})
-				</Box>
-			</UserCard.Info>
+				<AgentOrContactDetails mis='x10' name={username} status={<UserStatus status={status} />} />
+				{name && (
+					<Box display='flex' mis='x7' mb='x9' align='center' justifyContent='center'>
+						({name})
+					</Box>
+				)}
+			</Info>
```
Comment (ggazzo) [inline apps/meteor/ee/client/omnichannel/ContactManagerInfo.js:36]: do you agree with me that these component names all seem mismatched?
```
Info
	AgentOrContactDetails
	Box
```
should be something
```
AgentInfo
	AgentInfoDetails
	AgentInfoName

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/ee/client/omnichannel/ContactManagerInfo.js (hunk 2 above)
Comment (murtaza98) [reply apps/meteor/ee/client/omnichannel/ContactManagerInfo.js]: Alright makes sense. I've renamed that component. Please check

**Review by ggazzo — APPROVED** [review]

**Pull request merged**